### PR TITLE
Add "--include-full-xr" flag to the render command

### PR DESCRIPTION
**Pull request by ezgidemirel**
### Description of your changes

This PR adds a new flag `--include-full-xr` to the `render` command to output rendered composite resource's spec and metadata. This flag will be helpful with the `validate` command since it tries to validate XR spec as well. 

Example without the flag:
```
Ezgis-MacBook-Pro:validate ezgidemirel$ ./crossplane-cli beta render xr.yaml composition.yaml func.yaml 
---
apiVersion: example.crossplane.io/v1beta1
kind: XR
metadata:
  name: example
status:
  dummy: cool-status
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: AccessKey
metadata:
  annotations:
    crossplane.io/composition-resource-name: sample-access-key-0
  generateName: example-
  labels:
    crossplane.io/composite: example
  ownerReferences:
  - apiVersion: example.crossplane.io/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XR
    name: example
    uid: ""
spec:
  forProvider:
...
```

Example with the flag:
```
Ezgis-MacBook-Pro:validate ezgidemirel$ ./crossplane-cli beta render xr.yaml composition.yaml func.yaml --include-full-xr
---
apiVersion: example.crossplane.io/v1beta1
kind: XR
metadata:
  labels:
    a: b
  name: example
spec:
  count: 2
  id: abc
  maxReplicas: 3
status:
  dummy: cool-status
---
apiVersion: iam.aws.upbound.io/v1beta1
kind: AccessKey
metadata:
  annotations:
    crossplane.io/composition-resource-name: sample-access-key-0
  generateName: example-
  labels:
    crossplane.io/composite: example
  ownerReferences:
  - apiVersion: example.crossplane.io/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: XR
    name: example
    uid: ""
spec:
  forProvider:
...
```

Fixes # 

I have: 

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] ~Added or updated unit tests.~
- [ ] ~Added or updated e2e tests.~
- [ ] ~Linked a PR or a [docs tracking issue] to [document this change].~
- [ ] ~Added `backport release-x.y` labels to auto-backport this PR.~

Need help with this checklist? See the [cheat sheet].

[contribution process]: https://github.com/crossplane/crossplane/tree/master/contributing
[docs tracking issue]: https://github.com/crossplane/docs/issues/new
[document this change]: https://docs.crossplane.io/contribute/contribute
[cheat sheet]: https://github.com/crossplane/crossplane/tree/master/contributing#checklist-cheat-sheet

### CODE REVIEW

**Review by lsviben — APPROVED** [review]
LGTM. 

NIT: I prefer `--output-xr-spec` instead of `--include-xr-spec` but I dont feel strongly about it.

File: cmd/crank/beta/render/cmd.go [hunk 1]
```diff
@@ -45,6 +46,7 @@ type Cmd struct {
 	ContextFiles           map[string]string `mapsep:"," help:"Comma-separated context key-value pairs to pass to the Function pipeline. Values must be files containing JSON."`
 	ContextValues          map[string]string `mapsep:"," help:"Comma-separated context key-value pairs to pass to the Function pipeline. Values must be JSON. Keys take precedence over --context-files."`
 	IncludeFunctionResults bool              `short:"r" help:"Include informational and warning messages from Functions in the rendered output as resources of kind: Result."`
+	IncludeXRSpec          bool              `short:"x" help:"Include the XR spec to the rendered output."`
```
Comment (negz) [inline cmd/crank/beta/render/cmd.go:49]:
```suggestion
	IncludeXRSpec          bool              `short:"x" help:"Include a direct copy of the input XR spec in the rendered output."`
```

CC @plumbis for wordsmithing suggestions. I want to clarify succinctly that the spec is copied verbatim from the input you give. Unlike the rest of the output we're not actually rendering or computing it in any way.

**Review by negz — COMMENTED** [review]
@ezgidemirel Do you think there's a case to copy labels and annotations from the XR as well?

**Comment by negz** [comment]
> I prefer `--output-xr-spec` instead of `--include-xr-spec` but I dont feel strongly about it.

@lsviben I do too, except that we already have `--include-function-results`. I'm happy to go with `--output-...` if we feel it's worth renaming `--include-function-results` to `--output-function-results`.

**Comment by ezgidemirel** [comment]
> @ezgidemirel Do you think there's a case to copy labels and annotations from the XR as well?

@negz even though it's less likely to have metadata errors, I added them to catch cases like max label value length, or wrong type.

**Comment by ezgidemirel** [comment]
> LGTM.
> 
> NIT: I prefer `--output-xr-spec` instead of `--include-xr-spec` but I dont feel strongly about it.

I'm leaving it as it is until @plumbis comments.

File: cmd/crank/beta/render/cmd.go (hunk 1 above)
Comment (plumbis) [reply cmd/crank/beta/render/cmd.go]: I'd nix `direct` (what other kind of copy could it be?) but +1 on this suggestion.

**Comment by plumbis** [comment]
I'm not sure I fully understand what the flag does but `--include` implies "please add this other piece of data in the output" while `--output` makes it sound like you're doing something with only the XR. I think of `output` like `-o yaml` vs `-o json`

Do you have an example of the output with and without the flag enabled?

**Comment by ezgidemirel** [comment]
> I'm not sure I fully understand what the flag does but `--include` implies "please add this other piece of data in the output" while `--output` makes it sound like you're doing something with only the XR. I think of `output` like `-o yaml` vs `-o json`
> 
> Do you have an example of the output with and without the flag enabled?

@plumbis I updated the PR description with examples. The render command always contains an XR manifest. Without the flag, that manifest only contains the `metadata.name` and `status`. With the flag, we start to include labels, annotations and spec.

**Comment by plumbis** [comment]
Thanks for the added outputs @ezgidemirel ! 

With that I'd vote for `include` instead of `output`. Although it's an output you're asking the render command to `include` the additional information. 

I don't feel strongly about this and think both options are fine.

**Review by jbw976 — APPROVED** [review]
LGTM, I'm also good with the `--include-full-xr` flag name!

**Review by phisco — APPROVED** [review]

**Pull request merged**